### ISSUE TITLE
Comments from Eric

### ISSUE THREAD

**Issue by ericmjl** [hugobowne/deep-learning-from-scratch-pytorch#4]
Nonlinearity could be a good idea to explain with nothing more than a pithy line.

> At this point, we won't dive into which ones to use when but did wnt to highlight some common ones.

Hmm, there might be good rules-of-thumb. You could possibly make a simple example by considering the output layer:

- If the output needs to be squished between 0 and 1, use classic "sigmoid".
- If the output needs to be positive-only, use ReLU.
- If the output needs to be squashed between -1 and 1, use tanh.

Also, typo ("wnt").

> Use ReLU MLP now for 5 data points and one hidden layer with 8 nodes:

Nomenclature could be clarified, maybe? Hidden layer with 8 nodes refer to 8 output columns.

> Now we'll build a regressor with 4 hidden layers (setting $b=0$ throughout to simplify slightly):

Could be helpful to clarify that each layer has 3 nodes.

> 2.2 Back propagation

Section 2.2 is backprop, but at the end, the section ends with:

"Back prop calculates the slopes necessary to update the weights as it propagates back through the network. To calculate these slopes, we need use the chain rule from calculus, which is outside the scope of this workshop."

Going maybe a little bit deeper and showing regression loss w.r.t. "w" and "b" might be very eye-opening to the audience.

> Note that a round of forward prop and gradient descent is commonly called an epoch.

Epochs could be defined a bit more precisely, as it's a sloppily-used term. I think it refers to "one pass over the dataset" (this is contrasted most clearly against calculating data one batch at a time), not just one forward prop and gradient descent run. So for example, if I have 2000 rows, and I have batch size 50, then I have 40 iterations before I hit one epoch.

> HANDS ON: For many data points
It's now time to perform the same but for many data points! Instead of plotting the difference (which will be an array/vector), we'll plot the dot product of the difference with itself, which is a measure of distance.

I think I have two questions:

- This is equivalent to squared error, I think. Why not simply calling it squared error?
- Why not go further and calculate mean squared error?

I think your intended audience should be familiar enough to know this term.

**Comment by hugobowne** [comment]
@ericmjl this is all super helpful.

thanks for the nice heuristics/rules of thumb, nomenclature clarification, and the reminders about epochs & MSE!

**Comment by hugobowne** [comment]
covered everything here in https://github.com/hugobowne/deep-learning-from-scratch-pytorch/pull/6

thanks once again, Eric!
